core/src/main/java/org/jruby/util/OSEnvironment.java

```diff
@@ -111,17 +111,22 @@
             val = entry.getValue();
             if ( ! (val instanceof String) ) continue; // Java devs can stuff non-string objects into env
 
-            putRubyKeyValuePair(runtime, rubyMap, key, (String) val, encoding);
+            Encoding valueEncoding = keyEncoding;
+            if ( key.equals("PATH") ) {
```

Comment:
Mmm, I suppose this would be in line with environment vars being case-insensitive on Windows, wouldn't it.